### PR TITLE
tests: Add fuzzing harness for AS-mapping (asmap)

### PR DESCRIPTION
**Pull request by practicalswift**
Add fuzzing harness for AS-mapping (`asmap`).

To test this PR:

```
$ make distclean
$ ./autogen.sh
$ CC=clang CXX=clang++ ./configure --enable-fuzz \
      --with-sanitizers=address,fuzzer,undefined
$ make
$ src/test/fuzz/asmap
…
```

### CODE REVIEW

**Comment by MarcoFalke** [comment]
ACK 4d2aceaad8d28a54246b6639966e2278d2d795e3

File: src/netaddress.h [hunk 1]
```diff
@@ -39,7 +39,6 @@ class CNetAddr
         explicit CNetAddr(const struct in_addr& ipv4Addr);
         void SetIP(const CNetAddr& ip);
 
-    private:
```
Comment (jonatack) [inline src/netaddress.h]: `public:` at line 48 now redundant, can be removed as well?

**Review by jonatack — COMMENTED** [review]
ACK 4d2aceaad8d28a54246b6639966e2278d2d795e3

Code review, test. Fuzz output: https://gist.github.com/jonatack/5d78f682e0b245455192567d6af12b7e

PR #17812 adds test coverage as well (review beg).

**Comment by jonatack** [comment]
Tested #18023 for 18 hours and 72M execs with this fuzzer (before that patch it found a heap buffer overflow immediately, as shown in https://gist.github.com/jonatack/5d78f682e0b245455192567d6af12b7e):
```
#72474660 REDUCE cov: 1198 ft: 3529 corp: 197/12792b exec/s: 1063 rss: 452Mb L: 41/1527 MS: 4 InsertByte-EraseBytes-InsertByte-PersAutoDict- DE: "\x00 \x00\x00\x00\x00\x00\x00"-
```

**Pull request merged**